pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -179,10 +191,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 			c.Logger.Infof("PipelineRun %s's Pipeline %s can't be Run; it contains Tasks that don't exist: %s",
 				fmt.Sprintf("%s/%s", p.Namespace, p.Name),
 				fmt.Sprintf("%s/%s", p.Namespace, pr.Name), err)
-			// The PipelineRun is Invalid so we want to stop trying to Reconcile it
+			// This Run has failed, so we need to mark it as failed and stop reconciling it
+			pr.Status.SetCondition(&duckv1alpha1.Condition{
+				Type:   duckv1alpha1.ConditionSucceeded,
+				Status: corev1.ConditionFalse,
+			})
 			return nil
 		}
-		return fmt.Errorf("error getting Tasks and/or TaskRuns for Pipeline %s, Pipeline may be invalid!: %s", p.Name, err)
+		return fmt.Errorf("error getting Tasks and/or TaskRuns for Pipeline %s: %s", p.Name, err)
 	}
 	prtr := resources.GetNextTask(pr.Name, state, c.Logger)
```

Comment:
If there are no errors, set the condition to `Unknown` to indicate pipeline run is still running. Can also be done in the beginning of `reconcile()` before checking for errors

Reply:
i was going to do this, but it turned out we already had a test case covering it and it seemed to happen anyway! (i guess b/c of the state of the taskruns - but I'll initialize it anyway like you describe, just to be safe!